x-pack/plugins/security_solution/public/common/store/data_table/epic_local_storage.ts

```diff
@@ -34,7 +34,7 @@ const {
 
 export const isNotNull = <T>(value: T | null): value is T => value !== null;
 
-const tableActionTypes = [
+const tableActionTypes = new Set([
```

Comment:
The reason I changed all the epic filter action checks to using Sets is that they are saving a hundreds of unnecessary iterations. From reload to displaying a timeline, roughly 50 actions are dispatched.

Each epic receives all actions, so just the local-storage epic and the timeline-changed epic alone would in a worst case scenario need 1650 iterations when using array.includes(). Using Set, it's 100 in a worst case scenario.

Reply:
Sure thing, let me elaborate 👍  My point has to do with the amount of iterations that each epic's filter method has to do. So I counted how many redux actions are fired from the moment when the site loads until a timeline is show. That number is roughly 50.

The `local-storage-epic` and the `timeline-changed` epic (or previously the regular `epic.ts`) have a list of 13 and 20 actions that they respond to. For each of the 50 actions that are dispatched, these two epics then create 50 * 13 or 50 * 20 iterations to find elements in an array. That makes 1650 calls only to show a timeline.

Using a `Set`, which looks up elements in constant time (vs `Array`'s linear time), that lookup happens in 50 and 50 iterations because the set doesn't need to traverse its list of elements.

Reply:
Got it. thanks for the explanation. It is a great catch.